### PR TITLE
tech debt, style, and dependencies

### LINKED ISSUE #61: Setup psalm

### LINKED ISSUE #67: Replace lint.yml with split workflows
See the lint-*.yml files in https://github.com/nextcloud/.github/tree/master/workflow-templates

### PR DESCRIPTION
**Pull request by blizzz**
A collection of commits to make the app fit for NC 27. 

- chore: use Constructor property promotion
- fix: usage of deprecated interface
- fix: code style
- deps: update of (dev) dependencies
- ci: specify PHP platform
- ci: add psalm deps and config
- ci: split lint.yml

Resolves #61 
Resolves #67

### CODE REVIEW

File: lib/Flow/Operation.php [hunk 1]
```diff
@@ -44,30 +44,13 @@
...
-		IL10N $l,
-		IURLGenerator $urlGenerator,
-		IManager $notificationManager,
-		IUserSession $userSession,
-		LoggerInterface $logger
+		private IL10N $l,
+		private IURLGenerator $urlGenerator,
+		private IManager $notificationManager,
+		private IUserSession $userSession,
+		private LoggerInterface $logger
```
Comment (nickvergessen) [inline lib/Flow/Operation.php:52]:
```suggestion
		private LoggerInterface $logger,
```

For easier future, also allowed in php 8.0+

File: lib/Notification/Notifier.php [hunk 2]
```diff
@@ -37,19 +36,15 @@
...
-	private $urlGenerator;
 
-	public function __construct(IL10N $l, IURLGenerator $urlGenerator, IServerContainer $container) {
-		$this->l = $l;
-		$this->container = $container;
-		$this->urlGenerator = $urlGenerator;
+	public function __construct(
+		private IL10N $l,
+		private IURLGenerator $urlGenerator,
+		private ContainerInterface $container
```
Comment (nickvergessen) [inline lib/Notification/Notifier.php:46]:
```suggestion
		private ContainerInterface $container,
```

File: composer.json [hunk 3]
```diff
@@ -2,13 +2,22 @@
...
+	},
 	"scripts": {
-		"lint": "find . -name \\*.php -not -path './vendor/*' -print0 | xargs -0 -n1 php -l",
+		"lint": "find . -name \\*.php -not -path './vendor/*' -not -path './build/*' -not -path './node_modules/*' -print0 | xargs -0 -n1 php -l",
 		"cs:check": "php-cs-fixer fix --dry-run --diff",
-		"cs:fix": "php-cs-fixer fix"
+		"cs:fix": "php-cs-fixer fix",
+		"psalm": "psalm",
+		"psalm:fix": "psalm --alter --issues=InvalidReturnType,InvalidNullableReturnType,MissingParamType,InvalidFalsableReturnType",
+		"psalm:update-baseline": "psalm --update-baseline"
```
Comment (nickvergessen) [inline composer.json:21]: should commit the baseline file even when empty?

Comment (blizzz) [reply composer.json:21]: added

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**